src/ui/components/navigation/navigationcomponent.js

```diff
@@ -180,6 +180,10 @@ export default class NavigationComponent extends Component {
 
     this.checkOutsideClick = this.checkOutsideClick.bind(this);
     this.checkMobileOverflowBehavior = this.checkMobileOverflowBehavior.bind(this);
+
+    this.core.globalStorage.on('update', StorageKeys.API_CONTEXT, () => {
+      this.setState(this.getState());
```

Comment:
I'm not sure this will work. When we investigated yesterday, we found that for this component, setState is passed in globalStorage[StorageKeys.NAVIGATION], which is structured differently than the component's state.

Reply:
Ah forgot! I also definitely had a misunderstanding about the difference between globalStorage and getState. Changed all to use globalStorage (as that is the data from the /search/) so that we do not pass in extraneous data. However, could you expand on why you don't think this would work. I understand wanting to keep the input data format the same (thus changed), but it looks like the variables from getState would not override anything in the setState call. (Also tested and it was working locally)